src/test/scala/uk/gov/nationalarchives/tdr/api/service/FileServiceSpec.scala

```diff
@@ -805,6 +805,47 @@ class FileServiceSpec extends AnyFlatSpec with MockitoSugar with Matchers with S
     thrownException.getMessage should equal("No pagination input argument provided for 'paginatedFiles' field query")
   }
 
+  "getPaginatedFiles" should "return all the files by natural sorting order" in {
+    val fileRepositoryMock = mock[FileRepository]
+    val consignmentId = UUID.randomUUID()
+    val parentId = UUID.randomUUID()
+
+    val fileRowParams = List(
+      (UUID.randomUUID(), consignmentId, "fileName2", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName22", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName21", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName31", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName32", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName1", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName5", parentId),
+      (UUID.randomUUID(), consignmentId, "fileName3", parentId)
```

Comment:
Could you put a "fileName" in here without a digit so we can check it comes first in the list.

Reply:
Added it and it is working as expected after the change on the `NatuaralSorting` object as suggested in the above comment. Thanks.